web/shared/src/typeahead.ts

```diff
@@ -144,23 +152,47 @@ export function triage<T>(
             beginswithCaseSensitive.push(obj);
         } else if (lowerItem.startsWith(lowerQuery)) {
             beginswithCaseInsensitive.push(obj);
+        } else if (
+            lowerItem.includes(" " + lowerQuery) ||
+            lowerItem.includes("_" + lowerQuery) ||
+            lowerItem.includes("-" + lowerQuery) ||
+            lowerItem.includes("/" + lowerQuery)
```

Comment:
This could probably be written better with a regex, and a exported variable with the set of word boundaries supported; there's a good chance we'll end up extending the list.

`BEFORE_MENTION_ALLOWED_REGEX` is for a slightly different application in Markdown, but might be worth referencing when designing this detail.

Reply:
To involve regex I needed to import lodash into this file, but I think that's okay since other files in web/shared/src have imports (including lodash) too